gtsfm/averaging/rotation/cycle_consistency.py

```diff
@@ -227,23 +251,49 @@ def filter_to_cycle_consistent_edges(
     # (i1,i2) pairs
     cycle_consistent_keys = set()
 
+    per_edge_errors = defaultdict(list)
+
     triplets = extract_triplets(i2Ri1_dict)
 
     for (i0, i1, i2) in triplets:
         cycle_error, max_rot_error, max_trans_error = compute_cycle_error(
             i2Ri1_dict, (i0, i1, i2), two_view_reports_dict
         )
-
-        if cycle_error < CYCLE_ERROR_THRESHOLD:
-            # since i0 < i1 < i2 by construction, we preserve the property `a < b` for each edge (a,b)
-            cycle_consistent_keys.add((i0, i1))
-            cycle_consistent_keys.add((i1, i2))
-            cycle_consistent_keys.add((i0, i2))
+        # since i0 < i1 < i2 by construction, we preserve the property `a < b` for each edge (a,b)
+        per_edge_errors[(i0, i1)].append(cycle_error)
+        per_edge_errors[(i1, i2)].append(cycle_error)
+        per_edge_errors[(i0, i2)].append(cycle_error)
 
         cycle_errors.append(cycle_error)
         max_rot_errors.append(max_rot_error)
         max_trans_errors.append(max_trans_error)
 
+    errors_summary = []
+    errors_wrt_gt = []
+
+    plt.close("all")
+    # aggregate info over per edge_errors
+    for (i1, i2), edge_cycle_errors in per_edge_errors.items():
+        if edge_acceptance_criterion == EdgeErrorAggregationCriterion.MIN_EDGE_ERROR:
+            error_summary = np.amin(edge_cycle_errors)
```

Comment:
I don’t think error_summary is a good name, maybe aggregate_cycle_error ?

Reply:
hmm seems a bit too long IMHO

Reply:
Summary does not seem like a scalar, used the word ‘aggregate’ as we have AggregationCriterion.

Reply:
maybe `error_summary_stat` or `error_summary_val` or `aggregated_error`? which of those would you prefer @akshay-krishnan ?